src/Microsoft.ApplicationInsights/DataContracts/DependencyTelemetry.cs

```diff
@@ -122,6 +125,14 @@ public override string Id
             set { this.InternalData.id = value; }
         }
 
+        /// <summary>
+        /// Gets the dependency operation details, if any.
+        /// </summary>
+        public IDictionary<string, object> OperationDetails
```

Comment:
I'm concerned with two things here

1. confusion - we have too many things already.  I imagine users would not understand what info goes here and is it sent to AI backend.
2. abuse - users should not use this dict to pass something unrelated

One possible way to address at least first could be to
1. Make operation detail private field
2. Add getter and setter functions 
3. mark setter function as not browsable (like [TelemetryClient.Track](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/src/Microsoft.ApplicationInsights/TelemetryClient.cs#L432))
4. keep getter public and browsable and give a detailed documentation summary of what it is.

Another way 
1. keep `OperationDetails` public, but not browsable
2. Add extension method in the DependencyCollector package like `TryGetValue(string key)`

Even more strict way could be make an enum with possible keys in the DependenycCollector package and allow to get by enum.